cmd/gb-vendor/fetch.go

```diff
@@ -56,7 +56,11 @@ Flags:
 			return err
 		}
 
-		wc, err := repo.Checkout(branch, revision)
+		if m.HasDependencyWithUrl(repo.URL()) {
+			return fmt.Errorf("specified repository is already downloaded")
```

Comment:
Something like `fmt.Errorf("%s is already vendored from %s repository", path, repo.URL())` ?